### ISSUE TITLE
Error: curl::curl_fetch_memory(x$url$url, handle = x$url$handle) 

### ISSUE THREAD

**Issue by paternogbc** [ropensci/taxize#726]
Helo, I am getting an error every time I try to run `get_uid` or `get_taxa` for a long list of species.

```r
ud <- get_uid(sciname = ord, rank_query = "family")
Error in curl::curl_fetch_memory(x$url$url, handle = x$url$handle) : 
  Error in the HTTP2 framing layer
```
<details> <summary><strong>Session Info</strong></summary>

```r
> devtools::session_info()
─ Session info ──────────────────────────────────────────────────────────────────────────────────
 setting  value                       
 version  R version 3.4.4 (2018-03-15)
 os       Ubuntu 18.04.1 LTS          
 system   x86_64, linux-gnu           
 ui       RStudio                     
 language en_US                       
 collate  en_US.UTF-8                 
 ctype    pt_BR.UTF-8                 
 tz       America/Sao_Paulo           
 date     2018-12-16                  

─ Packages ──────────────────────────────────────────────────────────────────────────────────────
 ! package     * version date       lib source                                 
   ape           5.2     2018-09-24 [1] CRAN (R 3.4.4)                         
   assertthat    0.2.0   2017-04-11 [1] CRAN (R 3.4.4)                         
   backports     1.1.2   2017-12-13 [1] CRAN (R 3.4.4)                         
   base64enc     0.1-3   2015-07-28 [1] CRAN (R 3.4.4)                         
   bindr         0.1.1   2018-03-13 [1] CRAN (R 3.4.4)                         
   bindrcpp      0.2.2   2018-03-29 [1] CRAN (R 3.4.4)                         
   bit           1.1-14  2018-05-29 [1] CRAN (R 3.4.4)                         
   bit64         0.9-7   2017-05-08 [1] CRAN (R 3.4.4)                         
   blob          1.1.1   2018-03-25 [1] CRAN (R 3.4.4)                         
   bold          0.8.0   2018-10-26 [1] CRAN (R 3.4.4)                         
   callr         3.0.0   2018-08-24 [1] CRAN (R 3.4.4)                         
   cli           1.0.1   2018-09-25 [1] CRAN (R 3.4.4)                         
   codetools     0.2-15  2016-10-05 [4] CRAN (R 3.4.2)                         
   colorspace    1.3-2   2016-12-14 [1] CRAN (R 3.4.4)                         
   crayon        1.3.4   2017-09-16 [1] CRAN (R 3.4.4)                         
   crul          0.6.0   2018-07-10 [1] CRAN (R 3.4.4)                         
   curl          3.2     2018-03-28 [1] CRAN (R 3.4.4)                         
   data.table    1.11.8  2018-09-30 [1] CRAN (R 3.4.4)                         
   datastorr     0.0.3   2018-12-16 [1] Github (ropenscilabs/datastorr@0af2f0c)
   DBI           1.0.0   2018-05-02 [1] CRAN (R 3.4.4)                         
   dbplyr        1.2.2   2018-07-25 [1] CRAN (R 3.4.4)                         
   desc          1.2.0   2018-05-01 [1] CRAN (R 3.4.4)                         
   devtools    * 2.0.1   2018-10-26 [1] CRAN (R 3.4.4)                         
   digest        0.6.18  2018-10-10 [1] CRAN (R 3.4.4)                         
   dplyr       * 0.7.8   2018-11-10 [1] CRAN (R 3.4.4)                         
   foreach       1.4.4   2017-12-12 [1] CRAN (R 3.4.4)                         
   fs            1.2.6   2018-08-23 [1] CRAN (R 3.4.4)                         
   ggplot2       3.1.0   2018-10-25 [1] CRAN (R 3.4.4)                         
   ggtree        1.10.5  2018-12-15 [1] Bioconductor                           
   glue          1.3.0   2018-07-17 [1] CRAN (R 3.4.4)                         
   gtable        0.2.0   2016-02-26 [1] CRAN (R 3.4.4)                         
   hoardr        0.5.2   2018-12-02 [1] CRAN (R 3.4.4)                         
   httpcode      0.2.0   2016-11-14 [1] CRAN (R 3.4.4)                         
 V httr          1.3.1   2018-12-11 [1] CRAN (R 3.4.4)                         
   iterators     1.0.10  2018-07-13 [1] CRAN (R 3.4.4)                         
   jsonlite      1.6     2018-12-07 [1] CRAN (R 3.4.4)                         
   lattice       0.20-35 2017-03-25 [4] CRAN (R 3.4.2)                         
   lazyeval      0.2.1   2017-10-29 [1] CRAN (R 3.4.4)                         
   magrittr      1.5     2014-11-22 [1] CRAN (R 3.4.4)                         
   memoise       1.1.0   2017-04-21 [1] CRAN (R 3.4.4)                         
   munsell       0.5.0   2018-06-12 [1] CRAN (R 3.4.4)                         
   nlme          3.1-131 2017-02-06 [4] CRAN (R 3.4.2)                         
   packrat       0.5.0   2018-11-14 [1] CRAN (R 3.4.4)                         
   pbapply     * 1.3-4   2018-01-10 [1] CRAN (R 3.4.4)                         
   pillar        1.3.0   2018-07-14 [1] CRAN (R 3.4.4)                         
   pkgbuild      1.0.2   2018-10-16 [1] CRAN (R 3.4.4)                         
   pkgconfig     2.0.2   2018-08-16 [1] CRAN (R 3.4.4)                         
   pkgload       1.0.2   2018-10-29 [1] CRAN (R 3.4.4)                         
   plyr          1.8.4   2016-06-08 [1] CRAN (R 3.4.4)                         
   prettyunits   1.0.2   2015-07-13 [1] CRAN (R 3.4.4)                         
   processx      3.2.0   2018-08-16 [1] CRAN (R 3.4.4)                         
   ps            1.2.1   2018-11-06 [1] CRAN (R 3.4.4)                         
   purrr         0.2.5   2018-05-29 [1] CRAN (R 3.4.4)                         
   R6            2.3.0   2018-10-04 [1] CRAN (R 3.4.4)                         
   rappdirs      0.3.1   2016-03-28 [1] CRAN (R 3.4.4)                         
   Rcpp          1.0.0   2018-11-07 [1] CRAN (R 3.4.4)                         
   remotes       2.0.2   2018-10-30 [1] CRAN (R 3.4.4)                         
   reshape       0.8.8   2018-10-23 [1] CRAN (R 3.4.4)                         
   reshape2      1.4.3   2017-12-11 [1] CRAN (R 3.4.4)                         
   ritis         0.7.2   2018-05-20 [1] CRAN (R 3.4.4)                         
   rlang         0.3.0.1 2018-10-25 [1] CRAN (R 3.4.4)                         
   RMySQL        0.10.15 2018-05-13 [1] CRAN (R 3.4.4)                         
   RPostgreSQL   0.6-2   2017-06-24 [1] CRAN (R 3.4.4)                         
   rprojroot     1.3-2   2018-01-03 [1] CRAN (R 3.4.4)                         
   RSQLite       2.1.1   2018-05-06 [1] CRAN (R 3.4.4)                         
   rstudioapi    0.8     2018-10-02 [1] CRAN (R 3.4.4)                         
   rvcheck       0.1.3   2018-12-06 [1] CRAN (R 3.4.4)                         
   scales        1.0.0   2018-08-09 [1] CRAN (R 3.4.4)                         
   sessioninfo   1.1.1   2018-11-05 [1] CRAN (R 3.4.4)                         
   solrium       1.0.2   2018-12-13 [1] CRAN (R 3.4.4)                         
   storr         1.2.1   2018-10-18 [1] CRAN (R 3.4.4)                         
   stringi       1.2.4   2018-07-20 [1] CRAN (R 3.4.4)                         
   stringr       1.3.1   2018-05-10 [1] CRAN (R 3.4.4)                         
   taxize      * 0.9.4   2018-07-24 [1] CRAN (R 3.4.4)                         
   taxizedb    * 0.1.4   2017-06-20 [1] CRAN (R 3.4.4)                         
   taxonlookup * 1.1.5   2018-12-16 [1] Github (wcornwell/taxonlookup@a50aaff) 
   Taxonstand  * 2.1     2017-11-02 [1] CRAN (R 3.4.4)                         
   testthat      2.0.1   2018-10-13 [1] CRAN (R 3.4.4)                         
   tibble        1.4.2   2018-01-22 [1] CRAN (R 3.4.4)                         
   tidyr         0.8.2   2018-10-28 [1] CRAN (R 3.4.4)                         
   tidyselect    0.2.5   2018-10-11 [1] CRAN (R 3.4.4)                         
   treeio        1.2.2   2018-12-15 [1] Bioconductor                           
   triebeard     0.3.0   2016-08-04 [1] CRAN (R 3.4.4)                         
   urltools      1.7.1   2018-08-03 [1] CRAN (R 3.4.4)                         
   usethis     * 1.4.0   2018-08-14 [1] CRAN (R 3.4.4)                         
   withr         2.1.2   2018-03-15 [1] CRAN (R 3.4.4)                         
   xml2          1.2.0   2018-01-24 [1] CRAN (R 3.4.4)                         
   zoo           1.8-4   2018-09-19 [1] CRAN (R 3.4.4)                         

[1] /home/paterno/R/x86_64-pc-linux-gnu-library/3.4
[2] /usr/local/lib/R/site-library
[3] /usr/lib/R/site-library
[4] /usr/lib/R/library

 V ── Loaded and on-disk version mismatch.
```
</details>

**Comment by sckott** [comment]
can you give a reproducible example that fails?

**Comment by matteodelucchi** [comment]
I got the same error with:
`> classification(4097, db="ncbi")`
resulting in:
`Error in curl::curl_fetch_memory(x$url$url, handle = x$url$handle) : 
  Error in the HTTP2 framing layer`

However the error doesn't  always appear: 
When I try with an other id:
`> classification(4072, db="ncbi")`
it works. 

When I wait for a bit and try it again with the first id, it works...

The error is not depending on the id. I tried different id's and sometimes the error pops up, sometimes it doesn't (makes it hard to reproduce...)

Systeminfo:
`> version`
               _                           
platform       x86_64-pc-linux-gnu         
arch           x86_64                      
os             linux-gnu                   
system         x86_64, linux-gnu           
status                                     
major          3                           
minor          5.2                         
year           2018                        
month          12                          
day            20                          
svn rev        75870                       
language       R                           
version.string R version 3.5.2 (2018-12-20)
nickname       Eggshell Igloo 


Let me know if I can provide more information...

**Comment by sckott** [comment]
thanks @matteodelucchi - session info should also at least include the version of the package in question (taxize) 

seems from https://github.com/jeroen/curl/issues/156 that ideally its a server fix, but we don't control that, so possibly this https://github.com/jeroen/curl/issues/156#issuecomment-410241974 where you can set a curl option, with taxize that would be e.g., `classification(4072, db="ncbi", callopts=list(http_version = 0L))`

**Comment by matteodelucchi** [comment]
I use
`packageVersion("taxize")`
[1] ‘0.9.6’

When I disable http/2 it seems to work!
Thank you for your help!

**Comment by joelnitta** [comment]
I posted a comment at https://github.com/ropensci/rentrez/issues/127 that I think is relevant to this issue. 

tldr: you also might want to try `httr::set_config(httr::config(http_version = 0))` before using `classification()` if the above fix doesn't work.

**Comment by dwinter** [comment]
Hi all, if this is still an issue for anyone I think the latest version of `rentrez` (v1.2.2 bouncing around CRAN mirrors now) will have sorted this problem. It's possible some of the work-arounds people have used will produce warning messages, because `rentrez` is setting the `http_version` option to match the NCBI sever (and `warning` if this overwrites another config)

**Comment by sckott** [comment]
thanks @dwinter
